### ISSUE TITLE
Returning Date from null_value is useless

### ISSUE THREAD

**Issue by KodrAus** [KodrAus/elasticsearch-rs#147]
Take the following example:

```rust
struct MyDateMapping;
impl DateMapping for MyDateMapping {
    fn null_value() -> Date<Self::Format, Self> {
        Date::now()
    }
}
```

This won't do what you'd expect. The date value will be resolved when the mapping is serialised, and that's the date that will always be used. We should support date math in this null value, so we can set the null value to `now`.

This could look something like:

```rust
fn null_value() -> DateType<Self::Format, Self> {
    DateType::Math(DateMath::Now)
}
```

where `DateType` looks something like:

```rust
enum DateType<F: DateFormat, M: DateMapping<Format = F>> {
    Value(Date<F, M>),
    Math(DateMath)
}
```

The structure should be serialisable. We should make it as easy as possible to work with dates that are either math or values.

**Comment by KodrAus** [comment]
Depends on #29
